### ISSUE TITLE
[ISSUE] Create function returns error code missing required field

### ISSUE THREAD

**Issue by johnayoub** [databricks/databricks-sdk-py#413]
**Description**
Trying to create a function via the API or the SDK

**Reproduction**
/api/2.1/unity-catalog/functions

```json
{
  "catalog_name": "demo_catalog",
  "data_type": "STRING",
  "full_data_type": "STRING",
  "is_deterministic": true,
  "name": "demo_function",
  "routine_body": "SQL",
  "routine_definition": "'Hello World!'",
  "schema_name": "demo_schema",
  "sql_data_access": "CONTAINS_SQL"
}
```

**Expected behavior**
Function to be created

**Debug Logs**
```json
{
    "error_code": "INVALID_PARAMETER_VALUE",
    "message": "Missing required field: function_info",
    "details": [
        {
            "@type": "type.googleapis.com/google.rpc.RequestInfo",
            "request_id": "7a7855d8-de85-4de0-8a95-232b84bf5822",
            "serving_data": ""
        }
    ]
}
```

**Other Information**
None

**Additional context**
This fails whether using the API directly or the SDK

**Comment by adamcain-db** [comment]
Problem is the API is expecting the FunctionInfo to be the value of a `"function_info"` field (i.e., it is not 'inline').  Will fix the OpenAPI spec.  Try input that looks something like:
```
{
  "function_info": {
    "catalog_name": "demo_catalog",
    "data_type": "STRING",
    "full_data_type": "STRING",
    "is_deterministic": true,
    "name": "demo_function",
    "routine_body": "SQL",
    "routine_definition": "'Hello World!'",
    "schema_name": "demo_schema",
    "sql_data_access": "CONTAINS_SQL",
    "is_null_call": false,
    "parameter_style": "S",
    "security_type": "DEFINER",
    "specific_name": "some_name"
  }
}
```
Note the four added fields at the end.

**Comment by johnayoub** [comment]
thanks @adamcain-db. You're right the spec needs updating. I tried this via the API and the function was created. However, when I tried calling the function, I got the following error. I also can't seem to be able to delete the function via SQL. Deleting via the API works though.

`No content to map due to end-of-input
 at [Source: ; line: 1, column: 1]`

**Comment by adamcain-db** [comment]
@johnayoub: Thanks for confirming that creation worked for you with the modified input.  Not sure why deletion via SQL didn't work or why calling the function failed (separate ticket?).  We'll work on correcting the OpenAPI spec and the SDK.

**Comment by johnayoub** [comment]
@adamcain-db thanks! will open a separate ticket.

**Comment by johnayoub** [comment]
@adamcain-db closing as this was addressed in version 0.13.0
